lib/Myriad/Service/Implementation.pm

```diff
@@ -93,7 +93,7 @@ Populate internal configuration.
 =cut
 
 method configure (%args) {
-    $service_name = delete $args{name} if exists $args{name};
+    $service_name = delete $args{name} // die 'need a name';
```

Comment:
can't do this, `->configure` can be called multiple times - put it in `_add_to_loop` or `_init` maybe?